src/core/model.jl

```diff
@@ -101,32 +128,43 @@ end
 """
     agent_validator(agent, space)
 Validate the user supplied agent (subtype of `AbstractAgent`).
-Checks for mutability and existance and correct types for fields depending on `SpaceType`.
+Checks for mutability and existence and correct types for fields depending on `SpaceType`.
 """
 function agent_validator(::Type{A}, space::S) where {A<:AbstractAgent, S<:SpaceType}
     # Check A for required properties & fields
-    isconcretetype(A) || throw(ArgumentError("Agent struct must be a concrete type"))
-    isbitstype(A) && throw(ArgumentError("Agent struct must be mutable"))
-    (any(isequal(:id), fieldnames(A)) && fieldnames(A)[1] == :id) || throw(ArgumentError("First field of Agent struct must be `id`"))
+    if VERSION < v"1.1"
+        isconcrete = true
+        for f in 1:fieldcount(A)
+            if !isconcretetype(fieldtype(A, f))
+                isconcrete = false
+                break
+            end
+        end
+    else
+        iscontrete = all(isconcretetype, fieldtypes(A))
+    end
+    isconcrete || throw(ArgumentError("Agent struct must be a concrete type. If your agent is parametrically typed, this probably happened because you gave `Agent` instead of `Agent{T}` to this function. You can also create and instance of your agent and pass it to this function."))
```

Comment:
Here instead of `Agent{T}` write `Agent{Float64}`. `Agent{T}` is ambiguous depending on `T`

Reply:
Typo: `and instance` -> `an instance`